linetools/spectra/xspectrum1d.py

```diff
@@ -963,6 +963,97 @@ def stitch(self, idx=None, scale=1.):
         # Return
         return spec
 
+    def get_local_s2n(self, wv0, npix=51, flux_th=0., debug=False):
+        """It computes the local average signal-to-noise (s2n) over npix pixels around wv0.
+        If `flux_th` is given, pixels with fluxes below spec.flux*flux_th are masked out.
+
+        Parameters
+        ----------
+        wv0 : Quantity
+            Observed wavelength where to perform the calculation.
+        npix : float or int, optional
+            Number of pixels to perform the calculation (does not
+            take into account those pixel masked when fl_th is given).
+            Forced to be an odd int. Default is 51.
+        flux_th : float or array of same dimension as self.flux, optional
+            Minimum flux threshold for the S/N estimation. In other words,
+            pixels with self.flux < flux_th are masked out. This is useful
+            to exclude strong absorption features that may be present.
+            If flux_th is float then two things can happen:
+                a. If continuum is defined, we mask self.flux < self.co * flux_th
+                b. If continuum is not defined, we approximate the continuum by smoothing
+                   the spectrum convolving it with a Gaussian kernel of FWHM (in pixels) given
+                   by the maximum between 10*npix and 500.
+
+        Returns
+        -------
+        s2n : float
+            Local average signal-to-noise
+        s2n_sig : float
+            Standard deviation of the s2n measurement
+        """
+
+        # Do some checks
+        if (wv0 < self.wvmin) or (wv0 > self.wvmax):
+            raise ValueError("`wv0` is outside spectral range.")
+        if not self.sig_is_set:
+            raise ValueError("Spectrum has not defined an error array; cannot compute signal-to-noise.")
+        npix = int(npix)
+
+        # renaming
+        wa = self.wavelength
+        fl = self.flux
+        er = self.sig
+
+        # if flux_th is float, then check whether continuum exists
+        # otherwise we estimate it.
+        if isinstance(flux_th, float):
+            if self.co_is_set:
+                flux_th = self.co * flux_th
+            else:
```

Comment:
Add a comment here as to what you are doing.  Looks like you are generating a local continuum.